src/jit/lower.cpp

```diff
@@ -2862,8 +2862,8 @@ void Lowering::InsertPInvokeMethodProlog()
     firstBlockRange.InsertBefore(insertionPoint, LIR::SeqTree(comp, store));
     DISPTREERANGE(firstBlockRange, store);
 
-#ifndef _TARGET_X86_ // For x86, this step is done at the call site (due to stack pointer not being static in the
-                     // function).
+#if !defined(_TARGET_X86_) && !defined(_TARGET_ARM_)
+    // For x86 and arm32, this step is done at the call site (due to stack pointer not being static in the function).
 
```

Comment:
Ok, I see that your #ifdefs are correct, as this is not needed for ARM32 because it is set up by the call to `CORINFO_HELP_INIT_PINVOKE_FRAME`. However, the comment is confusing: the reason we set RSP here for non-x86 is because RSP is static in the function. Can you update the comment to be correct for ARM32?

---
In reply to: [103531744](https://github.com/dotnet/coreclr/pull/9825#discussion_r103531744) [](ancestors = 103531744)